File: multilevel_inher.cpp
//multilevel inheritance in c++
#include<iostream>
using namespace std;
class class1{
    public :
    void showme(){
        cout<<"this is the base class function"<<endl;
    }
};
class class2:public class1{
    public :
    void showYou(){
        cout<<"This is second base class function"<<endl;

    }
};
class class3:public class2{
    public:
    void showUs(){
        cout<<"this is third class function"<<endl;

    }
};
main(){
    //creating object
    class3 c1;
    //calling methods 
    c1.showme();
    c1.showUs();
    c1.showYou();
}

File: multilevel_inher.dart
//multilevel inheritance in Dart
class Bird{  //this is the first class 
  void fly(){ 
    print("Bird is flying :)");
  }
}
class parrot extends Bird{  //second class inherites the first class
  void speak(){
    print("Parrot is speaking :)");
  }
}
class Eagle extends parrot{ //third class inherites the second class 
  void fly1(){
    print("Eagle is flying :)");
  }
}
void main(){
  //creating object
  Eagle e1=new Eagle();
  e1.fly();
  e1.speak();
  e1.fly1();
}